pkg/start/start.go

```diff
@@ -419,6 +429,13 @@ func (o *Options) NewControllerContext(cb *ClientBuilder) *Context {
 
 	sharedInformers := externalversions.NewSharedInformerFactory(client, resyncPeriod(o.ResyncInterval)())
 
+	// check to see if techpreview should be on or off.  If we cannot read the featuregate for any reason, it is assumed
+	// to be off.  If this value changes, the binary will shutdown and expect the pod lifecycle to restart it.
+	includeTechPreview := false
+	if gate, err := client.ConfigV1().FeatureGates().Get(context.TODO(), "cluster", metav1.GetOptions{}); err == nil {
+		includeTechPreview = gate.Spec.FeatureSet == configv1.TechPreviewNoUpgrade
+	}
```

Comment:
if `err != nil`, we should probably at least log it.  Or should it be fatal?  "Random network hiccup, crashing out the CVO, hopefully our replacement process will have better luck" sounds better to me than "random network hiccup, I guess I will ignore all those TechPreviewNoUpgrade manifests I had been keeping an eye on for as long as I survive as a process".

The `context.TODO` and lack of `error` return from `NewControllerContext` both make me feel that this is a slightly odd place to put the Kube-API call.  But the existing split between `Options.Run` and `Options.NewControllerContext` seems pretty arbitrary to me.  I'm about even between either plumbing `ctx` in and `error` out of `NewControllerContext`, or with pulling this Kube-API call up into `Options.Run`.  Thoughts?

Reply:
fixed in fda77fe, which makes the error fatal and shifts the API call up into `Run`; thanks :)